tests/functional/test_show.py

```diff
@@ -79,14 +79,46 @@ def test_find_package_not_found():
     assert len(list(result)) == 0
 
 
+def test_report_single_not_found(script):
+    """
+    Test that a single unmatched package is reported not found.
+    """
+    # The following should report an error as there no result to print hence
+    # expect_error=True which implies allow_stderr_warning=True
+    result = script.pip('show', 'abcd-3', expect_error=True)
+    assert 'WARNING: Package(s) not found:' in result.stderr
+    # Note that this should report the un-canonicalized name
+    assert 'abcd-3' in result.stderr
+    lines = result.stdout.splitlines()
+    assert len(lines) == 0
```

Comment:
This can just be `assert not lines` (or maybe `assert not result.stdout.splitlines()`).

Reply:
Changed